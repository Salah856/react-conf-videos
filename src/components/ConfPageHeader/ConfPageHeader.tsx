import * as cn from 'classnames';
import * as React from 'react';

import { Logo } from 'components';

import styles from './ConfPageHeader.scss';

type Props = {
  title: string;
  titleLink: string;
  tagline: string;
};

export const ConfPageHeader: React.FunctionComponent<Props> = ({
  title,
  titleLink,
  tagline
}) => (
  <div className={styles.header}>
    <Logo />
    <div className={styles.text}>
      <h1 className={styles.headerTitle}>
        <a href={titleLink} className={styles.link}>
          {title}
        </a>
      </h1>
      <p className={styles.headerText}> {tagline} </p>
      <p className={cn(styles.headerText, styles.contribute)}>
        Contribute
        <a
          className={styles.contributeLink}
          href="https://github.com/simonccarter/react-conf-videos"
        >
          here
        </a>
        .
      </p>
    </div>
  </div>
);
