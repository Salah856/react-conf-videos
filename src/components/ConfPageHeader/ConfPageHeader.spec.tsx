import * as React from 'react'

import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import { ConfPageHeader } from './ConfPageHeader'

describe('ConfPageHeader', () => {
  it('should render', () => {
    // arrange
    const props = {
      title: 'React Conf 2018',
      titleLink: 'www.alink.com',
      tagline: 'a tagline'
    }
    // act
    const comp = shallow(<ConfPageHeader {...props} />)

    // assert
    expect(toJSON(comp)).toMatchSnapshot()
  })
})
