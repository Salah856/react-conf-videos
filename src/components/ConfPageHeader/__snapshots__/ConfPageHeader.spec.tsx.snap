// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfPageHeader should render 1`] = `
<div
  className="header"
>
  <Component />
  <div
    className="text"
  >
    <h1
      className="headerTitle"
    >
      <a
        className="link"
        href="www.alink.com"
      >
        React Conf 2018
      </a>
    </h1>
    <p
      className="headerText"
    >
       
      a tagline
       
    </p>
    <p
      className="headerText contribute"
    >
      Contribute
      <a
        className="contributeLink"
        href="https://github.com/simonccarter/react-conf-videos"
      >
        here
      </a>
      .
    </p>
  </div>
</div>
`;
